### DISCUSSION TITLE
How to pass value to onInputChange when options is an object array?

### DISCUSSION THREAD

**Discussion by ghost** [JedWatson/react-select#4346] [Q&A] [answered]
SOMEWHERE, I found sample code on how to use an object array as the options source but I can't find it now.

Here's my code:

```
    <Select
      options={workOrders}
      getOptionLabel={wo => wo.Nbr}
      getOptionValue={wo => wo.Nbr}
      onInputChange={handleOnChange}
    />
```

I've also tried onChange. Same result (undefined).

This does not pass the selected value to the onInputChange handler. It says undefined.

I tried using inputValue, but can't figure out how to set it to Nbr in my object. the syntax used in getOptionLabel (and value) don't work with inputValue.

My object array has Nbr, Descr, Status as properties.

How can I get the selected value in the handler so I can do stuff with it?

**Comment by Rall3n** [comment, marked as answer]
You are using `onInputChange` which in combination with `inputValue` is used to manage the value of the search input. It is not used for the overall value of the component.

Instead you should use the `onChange` (and the `value`) prop.

The first argument of the `onChange` handler should be the object of the selected option.

```jsx
<Select
  options={workOrders}
  getOptionValue={...}
  getOptionLabel={...}
  onChange={option => console.log(option)}
/>
```

**Reply by ghost** [reply]
Awesome! Thank you so much!!
